libraries/AP_HAL_ChibiOS/RCOutput_bdshot.cpp

```diff
@@ -214,8 +236,10 @@ void RCOutput::bdshot_prepare_for_next_pulse(pwm_group& group)
             // no locking required
             group.bdshot.enabled = true;
         } else {
-            osalDbgAssert(!group.bdshot.ic_dma_handle[group.bdshot.curr_telem_chan]->is_locked(), "DMA handle is already locked");
-            group.bdshot.ic_dma_handle[group.bdshot.curr_telem_chan]->lock();
+            osalDbgAssert(!group.bdshot.curr_ic_dma_handle, "IC DMA handle has not been released");
+            group.bdshot.curr_ic_dma_handle = group.bdshot.ic_dma_handle[group.bdshot.curr_telem_chan];
+            osalDbgAssert(group.shared_up_dma || !group.bdshot.curr_ic_dma_handle->is_locked(), "IC DMA handle is already locked");
```

Comment:
could shared_up_dma be removed when not on iomcu dshot?

Reply:
Fixed